common-npm-packages/packaging-common/nuget/NuGetConfigHelper2.ts

```diff
@@ -64,76 +66,105 @@ export class NuGetConfigHelper2 {
         }
     }
 
-    public addSourcesToTempNuGetConfig(packageSources: IPackageSource[]): void
-    {
+    public addSourcesToTempNuGetConfig(packageSources: IPackageSource[]): void {
         tl.debug('Adding sources to nuget.config');
         this.ensureTempConfigCreated();
         this.addSourcesToTempNugetConfigInternal(packageSources);
     }
 
-    public setAuthForSourcesInTempNuGetConfig(): void
-    {
+    /**
+ * Set authentication for source in a temporary nuget config file.
+ *
+ *
+ * @param nugetVersion - Optional.  If nugetVersion is provided, will use it to check if version is equal to 4.9.2.  Nuget 4.9.2 has a bug that requires a special work around.  If nugetVersion is not provided, will assusm that nuget version is not 4.9.2.
+ * @returns void. This method set authentication for source in a temporary nuget config file.
+ *
+ */
+    public setAuthForSourcesInTempNuGetConfig(nugetVersion?: VersionInfoVersion): void {
```

Comment:
Awkward to treat dotnet and nuget 2 different ways. For NuGet, the caller is responsible for resolving the tools version, for dotnet, the callee resolves the versions. Resolve the nuget tools version in this function